### PR TITLE
Towards renaming ALF to CPC, alfc to ethos

### PR DESCRIPTION
**Pull request by ajreynol**
This is the first step towards renaming our default proof format to `cpc` and renaming the proof checker `alfc` to `ethos`.

The following things are TODO and will be done shortly after this PR is merged:
- [external] Update alfc repo link to ethos
- [external] Update alfc repo to naming conventions of Eunoia, update binary name.
- Update signatures *.smt3 to *.eo, directories in `proof/`
- Update `src/proof/alf` to `src/proof/eo`.
- Update cvc5 proof support to match new Eunoia format.

### CODE REVIEW

File: contrib/get-ethos-checker [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+$BASE_DIR/bin/ethos \$@
+
+EOF
+chmod +x $BASE_DIR/bin/ethos_check.sh
+
+popd
+
+echo ""
+echo "========== How to use EO =========="
+echo "Generate a EO proof with cvc5:"
```
Comment (aniemetz) [inline contrib/get-ethos-checker:83]:
```suggestion
echo "Generate an EO proof with cvc5:"
```

File: docs/proofs/output_cpc.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
+Proof format: Cooperating Proof Calculus
```
Comment (aniemetz) [inline docs/proofs/output_cpc.rst:1]: It looks like most of the text here is > 80 char per line.

File: docs/proofs/proofs.rst [hunk 3]
```diff
@@ -12,7 +13,7 @@ reasoning. cvc5 supports retrieving a proof object via the API
 Proof Rules
 ^^^^^^^^^^^
 
-A comprehensive description of the **proof rules** is found here:
+A comprehensive description of the **proof rules** of the Cooperating Proof Calculus can be found here:
```
Comment (aniemetz) [inline docs/proofs/proofs.rst:16]: ditto

File: docs/proofs/output_cpc.rst (hunk 2 above)
Comment (ajreynol) [reply docs/proofs/output_cpc.rst:1]: Addressed

File: contrib/get-ethos-checker [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+mkdir -p build && cd build
+cmake -DCMAKE_INSTALL_PREFIX="$BASE_DIR" ..
+make -j$(nproc)
+mkdir -p $BASE_DIR/bin
+cp ./src/alfc $BASE_DIR/bin/ethos
+popd
+
+##### signatures
+
+# The EO signatures live in the main cvc5 repository. We reference this
```
Comment (aniemetz) [inline contrib/get-ethos-checker:42]: Should this be CPC?

File: docs/proofs/output_cpc.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+========================================
+
+Using the flag :ref:`proof-format-mode=cpc <lbl-option-proof-format-mode>`, cvc5
```
Comment (aniemetz) [inline docs/proofs/output_cpc.rst:4]:
```suggestion
Using option :ref:`proof-format-mode=cpc <lbl-option-proof-format-mode>`, cvc5
```

File: docs/proofs/output_cpc.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+As a disclaimer, this means that it treats certain operators differently from
+the SMT-LIB standard.
+As an example, cvc5 uses mixed arithmetic internally, where integers and reals
+can appear together.
+A comprehensive list of these differences can be found in the Eunoia definition
+of CPC, as described below.
+
+An efficient C++ proof checker (Ethos) is available
+`here <https://github.com/cvc5/ethos>`_ which can check proofs in the CPC
+format.
```
Comment (aniemetz) [inline docs/proofs/output_cpc.rst:17]:
```suggestion
`Ethos <https://github.com/cvc5/ethos>`_, an efficient proof checker written in C++
allows to check proofs in the CPC format.
```

File: docs/proofs/output_cpc.rst [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+For a quick start, the cvc5 repository contains a
+:cvc5repo:`script <contrib/get-ethos-checker>` which will download and install
```
Comment (aniemetz) [inline docs/proofs/output_cpc.rst:19]:
```suggestion
:cvc5repo:`script <contrib/get-ethos-checker>` to download and install
```

File: contrib/get-ethos-checker (hunk 4 above)
Comment (ajreynol) [reply contrib/get-ethos-checker:42]: Yes, I'll fix.

File: contrib/get-ethos-checker [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+BASE_DIR=`pwd`
+mkdir -p $BASE_DIR/tmp/
+
+##### EO
+EO_DIR="$BASE_DIR/alf-checker"
+mkdir -p $EO_DIR
+
+# download and unpack ethos
+ETHOS_VERSION="204968e8cdd884782335481e4e7e967dc5becad9"
+download "https://github.com/cvc5/alfc/archive/$ETHOS_VERSION.tar.gz" $BASE_DIR/tmp/ethos.tgz
```
Comment (aniemetz) [inline contrib/get-ethos-checker:28]:
```suggestion
download "https://github.com/cvc5/ethos/archive/$ETHOS_VERSION.tar.gz" $BASE_DIR/tmp/ethos.tgz
```

File: contrib/get-ethos-checker [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+tar --strip 1 -xzf $BASE_DIR/tmp/ethos.tgz -C $EO_DIR
+
+# build and install Ethos
+pushd $EO_DIR
+mkdir -p build && cd build
+cmake -DCMAKE_INSTALL_PREFIX="$BASE_DIR" ..
+make -j$(nproc)
+mkdir -p $BASE_DIR/bin
+cp ./src/alfc $BASE_DIR/bin/ethos
```
Comment (aniemetz) [inline contrib/get-ethos-checker:37]: Is this still called `alfc`?

Comment (ajreynol) [reply contrib/get-ethos-checker:37]: Yes, I am doing this in stages to make the changes small: the commit references in this script is still for a version of the repo that builds `alfc`.  A followup PR to cvc5 (after this one) will update to a commit that builds `ethos` and then the signatures will be changed to the new `eo::` syntax.

**Review by mpreiner — APPROVED** [review]

**Pull request merged**